course3/homework-vector/dcollier_vector.cc

```diff
@@ -0,0 +1,102 @@
+// author: dancollier@optiver.com
+#include <iostream>
+#include <cassert>
+#include <cmath>
+
+using namespace std;
+
+class Vector
+{
+    // TODO
+    public:
+        Vector(){
+	     std::cout << "Default constructor, this=" << this << std::endl;
+	};
+	Vector(int argX, int argY) { 
+    	     std::cout << "Constructor(x,y), this=" << this << std::endl; 
+    	     x = argX;
+             y = argY;
+	};
+	~Vector(){
+             std::cout << "Freeing memory this=" << this << std::endl;
+	};
+
+	bool operator == (Vector &vec){
+	    return ( this->x == vec.GetX() && this->y == vec.GetY() );
+	} 
+	bool operator != (Vector &vec){
+	    if( this->x == vec.GetX() && this->y == vec.GetY() ){
+		return false;
+	    }
+	    else { return true; }
+	} 
+	Vector operator + (Vector &vec){
+	     int newX = this->x + vec.GetX() ;
+	     int newY = this->y + vec.GetY() ;	
+	     return Vector(newX,newY);
+	} 
+	friend ostream& operator << ( ostream& os, Vector &vec){
+	     return os << "Vector(" << vec.GetX() << ", "  << vec.GetY() << ")";
+	}
+
+	double GetX(){ return x;};
+	double GetY(){ return y;};
+        void SetX(int argX){ x = argX; }
+        void SetY(int argY){ y = argY; }
+	double GetRadius(){ return Radius(x,y) ; };
+	double GetTheta(){ return Theta(x,y) ; };
+
+   private:
+	double x;
+	double y;
+
+	double Radius(double x, double y){
```

Comment:
1. Why implementing it as a separate method?
2. If you are not accessing 'this' in a method, you can make it static